packages/ui/src/views/platform/types.ts

```diff
@@ -0,0 +1,33 @@
+export enum DirectionEnum {
+  PARENT = 'parent',
+  CHILD = 'child'
+}
+
+// Base properties that all entities must have
+export interface BaseEntityProps {
+  id: string
+  name: string
+}
+
+// Props for rendering a single entity item
+export interface EntityRendererProps<T extends BaseEntityProps> {
+  entity: T
+  isSelected: boolean
+  onSelect: (entity: T) => void
+}
+
+export interface ParentFolderRendererProps<S = string> {
+  parentFolder: S
+  onSelect: (scope: S) => void
+}
+
+export interface ChildFolderRendererProps<F = string> {
+  folder: F
+  onSelect: (folder: F) => void
+}
+
+export enum ScopeEnum {
+  ACCOUNT = 'account',
+  ORGANIZATION = 'organization',
+  PROJECT = 'project'
```

Comment:
this is specific to harness platform.. and it'll not be applicable to opensource. also not applicable for flex hierarchy. let's re-evaluate this if we can avoid adding scopes in the view itself.